.github/workflows/spark_master_test.yaml

```diff
@@ -42,9 +42,39 @@ jobs:
           sudo apt-get update
           sudo apt-get install -y make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git
           sudo apt install libedit-dev
+          curl -LO https://github.com/bufbuild/buf/releases/download/v1.28.1/buf-Linux-x86_64.tar.gz
+          mkdir -p ~/buf
+          tar -xvzf buf-Linux-x86_64.tar.gz -C ~/buf --strip-components 1
+          rm buf-Linux-x86_64.tar.gz
+          sudo apt install python3-pip --fix-missing
+          sudo pip3 install pipenv==2021.5.29
+          curl https://pyenv.run | bash
+          export PATH="~/.pyenv/bin:$PATH"
+          eval "$(pyenv init -)"
+          eval "$(pyenv virtualenv-init -)"
+          pyenv install 3.8.18
+          pyenv global system 3.8.18
+          pipenv --python 3.8 install
+          pipenv run pip install pyspark==3.5.0
+          pipenv run pip install flake8==3.5.0 pypandoc==1.3.3
+          pipenv run pip install black==23.9.1
+          pipenv run pip install importlib_metadata==3.10.0
+          pipenv run pip install mypy==0.982
+          pipenv run pip install mypy-protobuf==3.3.0
+          pipenv run pip install cryptography==37.0.4
+          pipenv run pip install twine==4.0.1
+          pipenv run pip install wheel==0.33.4
+          pipenv run pip install setuptools==41.1.0
+          pipenv run pip install pydocstyle==3.0.0
+          pipenv run pip install pandas==1.1.3
+          pipenv run pip install pyarrow==8.0.0
+          pipenv run pip install numpy==1.20.3
+          pipenv run pip install grpcio==1.62.0
+          pipenv run pip install grpcio-status==1.62.0
+          pipenv run pip install googleapis-common-protos==1.56.4
         if: steps.git-diff.outputs.diff
       - name: Run Spark Master tests
         # when changing TEST_PARALLELISM_COUNT make sure to also change it in spark_test.yaml
         run: |
-          TEST_PARALLELISM_COUNT=2 build/sbt -DsparkVersion=master "++ ${{ matrix.scala }}" clean spark/test
+          TEST_PARALLELISM_COUNT=2 IS_SPARK_MASTER=true pipenv run python run-tests.py --group connectServer
```

Comment:
you can't just remove the `spark/test` eh

Reply:
why can't you just add `connectServer/test` ?

Reply:
@scottsand-db I removed `spark/test` that the testing runs faster, I will add it back once I'm done.

Reply:
@scottsand-db `connectServer/test` Isn't this for Scala tests? I am trying to run Python tests?

Reply:
:dope: my bad